core/src/Instance/AdfsAuthority.cs

```diff
@@ -50,12 +50,8 @@ public AdfsAuthority(string authority, bool validateAuthority) : base(authority,
 
         protected override bool ExistsInValidatedAuthorityCache(string userPrincipalName)
         {
-            if (string.IsNullOrEmpty(userPrincipalName))
-            {
-                throw CoreExceptionFactory.Instance.GetClientException(
-                    CoreErrorCodes.UpnRequired,
-                    CoreErrorMessages.UpnRequiredForAuthroityValidation);
-            }
+            if (userPrincipalName == null)
```

Comment:
should use string.IsNullOrEmpty() here instead of just a null comparison against a string.